app/src/main/java/com/example/tasky/agenda/presentation/components/AgendaItem.kt

```diff
@@ -0,0 +1,238 @@
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.text.style.TextDecoration
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import com.example.tasky.R
+import com.example.tasky.agenda.presentation.model.Agenda
+import com.example.tasky.ui.theme.Green
+
+@Composable
+fun EventItem(
+    event: Agenda.Event,
+    onDoneClicked: (Boolean) -> Unit,
+    onMoreOptionsClicked: () -> Unit,
+    modifier: Modifier = Modifier
+) {
+    AgendaItem(
+        title = event.title,
+        description = event.description,
+        date = event.date,
+        onDoneClicked = onDoneClicked,
+        onMoreOptionsClicked = onMoreOptionsClicked,
+        modifier = modifier
+    )
+}
+
+@Composable
+fun ReminderItem(
+    reminder: Agenda.Reminder,
+    onDoneClicked: (Boolean) -> Unit,
+    onMoreOptionsClicked: () -> Unit,
+    modifier: Modifier = Modifier
+) {
+    AgendaItem(
+        title = reminder.title,
+        description = reminder.description,
+        date = reminder.date,
+        onDoneClicked = onDoneClicked,
+        onMoreOptionsClicked = onMoreOptionsClicked,
+        modifier = modifier
+    )
+}
+
+@Composable
+fun TaskItem(
+    task: Agenda.Task,
+    onDoneClicked: (Boolean) -> Unit,
+    onMoreOptionsClicked: () -> Unit,
+    modifier: Modifier = Modifier,
+    isDone: Boolean = false,
+) {
+    AgendaItem(
+        title = task.title,
+        description = task.description,
+        date = task.date,
+        onDoneClicked = onDoneClicked,
+        onMoreOptionsClicked = onMoreOptionsClicked,
+        isTask = true,
+        isDone = isDone,
+        modifier = modifier
+    )
+}
+
+@Composable
+fun AgendaItem(
+    title: String,
+    description: String,
+    date: String,
+    modifier: Modifier = Modifier,
+    isTask: Boolean = false,
+    isDone: Boolean = false,
+    onDoneClicked: (Boolean) -> Unit,
+    onMoreOptionsClicked: () -> Unit
+) {
+    Card(
+        modifier = modifier
+            .fillMaxWidth()
+            .padding(start = 14.dp, end = 8.dp),
+        colors = CardDefaults.cardColors(
+            containerColor = Green,
+            contentColor = Color.White
+        )
+    ) {
+        Column(
+            modifier = Modifier.padding(
+                start = 15.dp,
+                top = 17.dp,
+                end = 15.dp,
+                bottom = 12.dp
+            ),
+        ) {
+            Row(
+                modifier = Modifier.fillMaxWidth(),
+                horizontalArrangement = Arrangement.Center
+            ) {
+                if (isTask) {
+                    Column {
+                        Spacer(modifier = Modifier.size(5.dp))
+                        Icon(
+                            modifier = Modifier
```

Comment:
The icon with the circle, should be displayed on Task, Reminder and Event

But only the Task one should be clickable

To achieve this, you ould use a .then in the modifier of the Icon